fuel-vm/src/interpreter/blob.rs

```diff
@@ -0,0 +1,89 @@
+use fuel_asm::{
+    RegisterId,
+    Word,
+};
+use fuel_storage::StorageSize;
+use fuel_tx::PanicReason;
+use fuel_types::BlobId;
+
+use crate::{
+    error::IoResult,
+    prelude::*,
+    storage::{
+        BlobData,
+        InterpreterStorage,
+    },
+};
+
+use super::{
+    internal::inc_pc,
+    split_registers,
+    GetRegMut,
+    Interpreter,
+    Memory,
+    SystemRegisters,
+    WriteRegKey,
+};
+
+impl<M, S, Tx, Ecal> Interpreter<M, S, Tx, Ecal>
+where
+    M: Memory,
+    S: InterpreterStorage,
+    <S as InterpreterStorage>::DataError: From<S::DataError>,
+{
+    pub(crate) fn blob_size(
+        &mut self,
+        dst: RegisterId,
+        blob_id_ptr: Word,
+    ) -> IoResult<(), S::DataError> {
+        self.gas_charge(self.interpreter_params.gas_costs.bsiz().base())?;
+
+        let blob_id = BlobId::from(self.memory.as_ref().read_bytes(blob_id_ptr)?);
+
+        let Some(size) =
+            <S as StorageSize<BlobData>>::size_of_value(&self.storage, &blob_id)
+                .map_err(RuntimeError::Storage)?
+        else {
+            return Err(PanicReason::BlobNotFound.into());
+        };
+        self.dependent_gas_charge_without_base(
+            self.interpreter_params.gas_costs.bsiz(),
+            size as Word,
+        )?;
+        let (SystemRegisters { pc, .. }, mut w) = split_registers(&mut self.registers);
+        let result = &mut w[WriteRegKey::try_from(dst)?];
+        *result = size as Word;
```

Comment:
Naming the index would hurt readibility. The index cannot be used in any other context than when indexing into `ProgramRegisters`. The pattern `&mut program_registers[WriteRegKey::try_from(dst)?]` repeats everywhere, because that's the only sensible way to use `WriteRegKey`. Giving it a name would only be more confusing. That said, I'm pushing to get rid of `WriteRegKey` and `ProgramRegisters`, so we wouldn't have to do this.

Abbreviations are vital for code readibility. I cannot easily parse lines that have multiple long identifiers. Particularly things that get used all the time deserve a short name. The register names are used in every single instruction implementation, and having `program_counter` in every place where we have `pc` now would be awful. It's stored as `pc` in my head, I do not expand it every time.

`result` is the reference where the result is written to. What would be a better name? Maybe `result_register_value`?

The `w` is defined on the line above, and used once. Giving it a longer name would make things worse. Whice one you prefer?

```rust
let mut acc = 0;
for i in 0..limit {
  acc += values[i];
}
```

or

```rust
let mut accumulator = 0;
for values_index in 0..limit {
  accumulator += values[values_index];
}
```

Reply:
Sorry. I think I can give a more helpful recommendation.

Let's just write a helper that hides all the abbreviations from the reader:

Instead of 
```
        let (SystemRegisters { pc, .. }, mut w) = split_registers(&mut self.registers);
        let result = &mut w[WriteRegKey::try_from(dst)?];
        *result = size as Word;
```
just declaratively say

```
      self.write_value_to_result_register(size)
```

An order of magnitude easier to read.

>The w is defined on the line above, and used once. Giving it a longer name would make things worse. Whice one you prefer?

`acc` means accumulator to me. `i` means index. `w` means nothing.

I prefer the first in your example, but the second isn't painful for me. If there was any business context in the example, then I'd prefer the latter.

WRT readabiliy, I always prefer a declarative function over having to read a bunch of implementation details.